### PR TITLE
fix: use paranoid on groupedLimit.on

### PR DESCRIPTION
**Pull request by sushantdhiman**
Port for #6833 to master

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Future
-- [FIXED] `restore` now uses `field` from `deletedAt` 
+- [FIXED] Set `timestamps` and `paranoid` options from through model on `belongsToMany` association
+- [FIXED] Properly apply paranoid condition when `groupedLimit.on` association is `paranoid`
+- [FIXED] `restore` now uses `field` from `deletedAt`
```
Comment (janmeier) [inline changelog.md]: Is this actually fixed in master, or just a copy pasta?

Comment (sushantdhiman) [inline changelog.md]: its fixed for master as well, test will fail without said chanegs

**Comment by janmeier** [comment]
Hmm, fails for mysql :(

File: lib/associations/belongs-to-many.js [hunk 2]
```diff
@@ -171,6 +171,10 @@ class BelongsToMany extends Association {
       }
     }
 
+    this.options = _.assign(this.options, _.pick(this.through.model.options, [
```
Comment (felixfbecker) [inline lib/associations/belongs-to-many.js]: `Object.assign()`

File: test/integration/model/findAll/groupedLimit.test.js [hunk 3]
```diff
@@ -135,6 +161,55 @@ if (current.dialect.supports['UNION ALL']) {
...
             });
           });
+
+          it('works with paranoid junction models', function() {
+            return this.User.findAll({
+              attributes: ['id'],
+              groupedLimit: {
+                limit: 3,
+                on: this.User.ParanoidProjects,
+                values: this.projects.map((item) => item.get('id'))
```
Comment (felixfbecker) [inline test/integration/model/findAll/groupedLimit.test.js]: Unneeded parens (a few more times further down too)

**Comment by sushantdhiman** [comment]
@felixfbecker Do we really need these changes, there is no practical difference :), ping @janmeier

**Comment by felixfbecker** [comment]
It's included in the ESLint. Running `eslint --fix` would remove them all - there is already enough stuff to be fixed, I would like to at least keep new code additions clean. As for the justification of the rule itself, you could add hundreds of parenthesis around every expression in the code, but it makes the code less readable.

**Comment by sushantdhiman** [comment]
Hehe :smile_cat:

**Pull request merged**